### ISSUE TITLE
Saga not handling replies

### ISSUE THREAD

**Issue by tziemek** [artiso-solutions/adsd-hotel#99]
A problem occurs when the `ReservationSaga` is sending a command to another service and this service is responding with a reply, additionally to publishing an event. In this case there is an exception, because the `PurpleService` isn't handling messages for the type `Response<bool>`.
I am really uncomfortable with all the replies in our message handlers and currently see two solutions for this kind of problem:
1. Remove the replies together with the requests in the ambassadors. Commands should always succeed ;-)
2. Only do a reply if the command dto is really invalid and can't be processed. Atm there are a few

**Comment by tommasobertoni** [comment]
The replies were meant to be an optimization of a "unknown result" scenario, where otherwise the ambassadors would need to poll the backing service to check if the operation succeeded.
This scenario should only be applied when it's expected to there be a user on the other side of the operation, which may have just input some data that needs validation.

Replies are definitely not needed everywhere, and in some scenarios they're wrong, as you mentioned.

I think there may be a few points where we missed this and just went for request-reply as a default: we should identify them and fix the api.
More drastically, we may remove the replies altogether, but we'd need to implement the same behavior in some way, or maybe re-implement the replies only where needed, with a clean state?

I'd be inclined to go with option 2 and fix those points where we don't need a reply, because I see the advantages of them - in some scenarios.

**Comment by tziemek** [comment]
Yes, I also prefer option 2 atm, but we maybe we should think about the entry points to the system. In the black service for example, the external API is REST and not directly the messaging system. This process already handles validation problems because we can do a simple Req/Res with the WebAPI and only if all is valid we send out an event. In this scenario the calls to the WebAPI would be the commands (in an abstract manner), and we can ensure that the messaging system only receives valid messages. Therefore the commands are not `void` and it's an absolutely valid case.
After a bit of thinking of what commands are, I tend to take the opinion that commands don't have to be void, because a command always has two sides and receiving party can surely decline the message ("no, I don't like that" so to speak).

We already removed some of the validation, or rather changed a `ValidationException` to some other exception in the yellow service so the retries are triggered (see #100).
